gt_concordance/gt_concordance.py

```diff
@@ -0,0 +1,519 @@
+        click.echo('Analysis path does not exist! Path: {}'.format(ANALYSIS_PATH))
+        click.echo('Terminating')
+        exit(0)
+
+    XL_FILES_ARCHIVED = config.get('XL_FILES_ARCHIVED')
+    if XL_FILES_ARCHIVED is None:
+        click.echo('config file missing XL_FILES_ARCHIVED')
+        click.echo('Do you want to enter path where to archive excel files? Y/n')
+        to_enter = raw_input()
+        if to_enter.lower() == 'y':
+            XL_FILES_ARCHIVED = raw_input()
+        else:
+            click.echo('Please enter where to archive excel files')
+            XL_FILES_ARCHIVED = raw_input()
+    if not os.path.exists(XL_FILES_ARCHIVED):
+        click.echo('Path does not exist! Path: {}'.format(XL_FILES_ARCHIVED))
+        click.echo('Terminating')
+        exit(0)
+
+    click.echo('Looking for .xlsx files in {}'.format(XL_FILES_PATH))
+    xl_files = [os.path.join(XL_FILES_PATH, filename) for filename in os.listdir(XL_FILES_PATH) if '.xlsx' in filename]
+    if not xl_files:
+        click.echo('No .xlsx files found! Terminating')
+        exit(0)
+
+    click.echo('Following files are found: {}'.format(', '.join([os.path.basename(xl_file) for xl_file in xl_files])))
+
+    # parsing snps file
+    snps_data = parse_maf_snps_file(config)
+
+    # parsing xl files
+    genotype_data = {}
+    for xl_file in xl_files:
+        click.echo("Parsing file: {}".format(os.path.basename(xl_file)))
+        data = pyexcel_xlsx.get_data(xl_file)
+        data = data.get('HaploView_ped_0') # sheet name
+        # getting list of lists
+        header = data[0]
+        data = data[1:]
+        for row in data:
+            # row[1] is always sample name. If doesn't match NGI format - skip.
+            if not re.match(r"^ID\d+-P\d+_\d+", row[1]):
+                continue
+            # sample has format ID22-P2655_176
+            sample_id = row[1].split('-')[-1]
+            # if the same sample occurs twice in the same file, will be overwriten
+            if sample_id not in genotype_data:
+                genotype_data[sample_id] = {}
+            else:
+                click.echo('Sample {} has been already parsed from another (or this) file. Overwriting'.format(sample_id))
+            # rs positions start from 9 element. hopefully the format won't change
+            for rs_id in header[9:]:
+                rs_index = header.index(rs_id)
+                allele1, allele2 = row[rs_index].split()
+                genotype_data[sample_id][rs_id] = [allele1, allele2]
+
+    output_files = []
+    for sample_id in genotype_data:
+        project_id = sample_id.split('_')[0]
+        # create .gt file for each sample
+        output_path = os.path.join(ANALYSIS_PATH, project_id, 'piper_ngi/03_genotype_concordance')
+        if not os.path.exists(output_path):
+            click.echo('Output path does not exist! Path will be created: {}'.format(output_path))
+            os.makedirs(output_path)
```

Comment:
ok so in case the project has not yet a folder in ANALYSIS you create it. Yesterday we discuss to change this strategy and instead do not move to archive xls files containing samples belonging to projects not yet in our ANALYSIS. morevoer we said to add a comment line in the gt file (the first row) reporting the file name of the xls so that we can check for collisions and distinguish between new and already processed sampels